pkg/psalabelsyncer/privileged_namespaces_controller.go

```diff
@@ -0,0 +1,149 @@
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/controller"
+	psapi "k8s.io/pod-security-admission/api"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const privilegedControllerName = "privileged-namespaces-psa-label-syncer"
+
+type privilegedNamespacesPSALabelSyncer struct {
+	nsClient      corev1client.NamespaceInterface
+	nsLister      corev1listers.NamespaceLister
+	nsCacheSynced cache.InformerSynced
+
+	workqueue workqueue.RateLimitingInterface
+}
+
+func NewPrivilegedNamespacesPSALabelSyncer(
+	ctx context.Context,
+	namespaceClient corev1client.NamespaceInterface,
+	namespaceInformer corev1informers.NamespaceInformer,
+	eventRecorder events.Recorder,
+) *privilegedNamespacesPSALabelSyncer {
+	c := &privilegedNamespacesPSALabelSyncer{
+		nsClient: namespaceClient,
+		nsLister: namespaceInformer.Lister(),
+
+		nsCacheSynced: namespaceInformer.Informer().HasSynced,
+
+		workqueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), privilegedControllerName),
+	}
+
+	logger := klog.FromContext(ctx)
+
+	namespaceInformer.Informer().AddEventHandler(
+		cache.FilteringResourceEventHandler{
+			FilterFunc: factory.NamesFilter("default", "kube-system", "kube-public"),
+			Handler: cache.ResourceEventHandlerFuncs{
+				AddFunc: func(obj interface{}) {
+					c.enqueueNS(logger, obj)
+				},
+				UpdateFunc: func(_, newObj interface{}) {
+					c.enqueueNS(logger, newObj)
+				},
+				DeleteFunc: nil,
+			},
+		},
+	)
+
+	return c
+}
+
+func (c *privilegedNamespacesPSALabelSyncer) Run(ctx context.Context, _ int) {
+	defer utilruntime.HandleCrash()
+	defer c.workqueue.ShutDown()
+
+	logger := klog.FromContext(ctx)
+	logger.Info("Starting", "controller", privilegedControllerName)
+	defer logger.Info("Shutting down", "controller", privilegedControllerName)
+
+	if !cache.WaitForNamedCacheSync(privilegedControllerName, ctx.Done(), c.nsCacheSynced) {
+		return
+	}
+
+	go wait.UntilWithContext(ctx, c.worker, time.Second)
+
+	<-ctx.Done()
+}
+
+func (c *privilegedNamespacesPSALabelSyncer) worker(ctx context.Context) {
+	workerFunc := func(ctx context.Context) bool {
+		key, quit := c.workqueue.Get()
+		if quit {
+			return true
+		}
+		defer c.workqueue.Done(key)
+
+		logger := klog.FromContext(ctx)
+		logger = klog.LoggerWithValues(logger, "queueKey", key)
+		ctx = klog.NewContext(ctx, logger)
+
+		err := c.sync(ctx, key.(string))
+		if err == nil {
+			c.workqueue.Forget(key)
+			return false
+		}
+
+		utilruntime.HandleError(err)
+		c.workqueue.AddRateLimited(key)
+
+		return false
+	}
+
+	for {
+		if quit := workerFunc(ctx); quit {
+			klog.FromContext(ctx).Info("shutting down", "controller", privilegedControllerName)
```

Comment:
Won't this be logged twice because of [line 76](https://github.com/openshift/cluster-policy-controller/pull/128/files#diff-dada1af68a1323fc8f6385f3c7ba4e472badda0462047a4ece3ecde4a1176347R76)?

Reply:
true, I'll remove this bit